src/battle/battle_4.c

```diff
@@ -13003,20 +12737,17 @@ static void atkA3_disablelastusedattack(void)
 
 static void atkA4_trysetencore(void)
 {
-    int i;
-    for (i = 0; i < 4; i++)
+    u16 encoreMove = gLastUsedMetronomeMove[gBankTarget];
+    if (encoreMove == MOVE_NONE)
+        encoreMove = gLastUsedMove[gBankTarget];
+    if (encoreMove == MOVE_STRUGGLE || encoreMove == MOVE_ENCORE || encoreMove == MOVE_MIRROR_MOVE)
     {
-        if (gBattleMons[gBankTarget].moves[i] == gLastUsedMove[gBankTarget])
-            break;
+        gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 1);
```

Comment:
Where did this come from? What purpose does it serve? How does it relate to what you removed?